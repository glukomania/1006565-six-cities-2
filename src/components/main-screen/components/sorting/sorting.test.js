import renderer from 'react-test-renderer';
import Sorting from './sorting';

jest.mock(`react-router-dom`, () => ({
  Link: () => null
}));

it(`Sorting is displayed correctly`, () => {

  const changeHandle = jest.fn();
  const dropDownHandler = jest.fn();

  const sorting = renderer
    .create(<Sorting changeHandle={changeHandle} dropDownHandler={dropDownHandler}/>)
  .toJSON();

  expect(sorting).toMatchSnapshot();
});
