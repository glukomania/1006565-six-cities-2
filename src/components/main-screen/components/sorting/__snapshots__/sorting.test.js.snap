// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting is displayed correctly 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
   Â 
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom"
    onClick={[Function]}
  >
    <li
      className="places__option places__option--active"
      data-sorting="popular"
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option"
      data-sorting="priceLow"
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option"
      data-sorting="priceHigh"
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option"
      data-sorting="rated"
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
